### PR TITLE
Add script to deploy local canary build to dev environment

### LINKED ISSUE #2142: Add extra instructions to the devguide on installing a canary image
We should provide steps on how to do a local build with an alternative registry set, push it, and then update your deployment to use the canary image

e.g. 

```
make push REGISTRY=carolynvs/
helm upgrade --name catalog --set image=carolynvs/service-catalog:canary --set imagePullPolicy=Always
```

### PR DESCRIPTION
**Pull request by carolynvs**
This is what I used to make it easier for me to develop a new feature and deploy it locally to my cluster.

Doc preview at https://deploy-preview-2281--svc-cat.netlify.com/docs/devguide/#deploy-local-canary

Closes #2142

### CODE REVIEW

File: docs/devguide.md [hunk 1]
```diff
@@ -470,6 +470,21 @@ If you choose third party resources storage, the helm chart will not launch an
...
+For your convenience, you can use the following script quickly rebuild, push and
+deploy the canary image. There are a few assumptions about your environment and
+configuration in the script (for example, that you have persistent storage setup
+for etcd so that you don't lose data in between pushes). If the assumptions do
+not match your needs, we suggest copying the contents of that script and using
+it as a starting off point for your own custom deployment script.
+
+```console
+# The registry defaults to DockerHub with the same user name as the current user
+# Examples: quay.io/myuser/service-catalog/, anotherser/
```
Comment (shubheksha) [inline docs/devguide.md]: nit: anotherser is a typo?

File: charts/catalog/values.yaml [hunk 2]
```diff
@@ -4,6 +4,8 @@ image: quay.io/kubernetes-service-catalog/service-catalog:v0.1.31
 # imagePullPolicy for the service-catalog; valid values are "IfNotPresent",
 # "Never", and "Always"
 imagePullPolicy: Always
+# deploymentStrategy for service-catalog; value values are "RollingUpdate" and "Recreate"
```
Comment (shubheksha) [inline charts/catalog/values.yaml]: Have we documented both the deployment strategies and the differences somewhere?

Comment (jboyd01) [reply charts/catalog/values.yaml]: @shubheksha: It's a kube term, https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy

File: docs/devguide.md (hunk 1 above)
Comment (carolynvs) [reply docs/devguide.md]: yup!

Comment (carolynvs) [reply docs/devguide.md]: Fixed

**Comment by carolynvs** [comment]
Dear CLA Bot, please recheck me for a CLA, kthnxbye

**Pull request merged**